docs/DeveloperGuide.md

```diff
@@ -5,9 +5,238 @@
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 ## Design & implementation
+### Adding calorie entries feature
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Implementation
```

Comment:
Good that you zoomed into implementation via the header. What happened to the Design part?